application-analytics-ui/src/main/resources/Analytics/Code/Macros/RowEvolutionJson.xml

```diff
@@ -0,0 +1,54 @@
+<?xml version="1.1" encoding="UTF-8"?>
+
+<!--
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc version="1.4" reference="Analytics.Code.Macros.RowEvolutionJson" locale="">
+  <web>Analytics.Code.Macros</web>
+  <name>RowEvolutionJson</name>
+  <language/>
+  <defaultLanguage/>
+  <translation>0</translation>
+  <creator>xwiki:XWiki.Admin</creator>
+  <parent>WebHome</parent>
+  <author>xwiki:XWiki.Admin</author>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <version>1.1</version>
+  <title>RowEvolutionJson</title>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>true</hidden>
+  <content>{{velocity}}
+#set ($parameters ={
+  'period' : $request.period,
+  'date' : $request.date,
+  'module' : 'API',
+  'method' : 'Actions.getPageUrl',
+  'format' : 'json'
+})
+#if ($request.macroName=='MostViewedPages')
+  $parameters.put('pageUrl', $request.parameter)
+#end
+#set ($analyticsResult = $services.analytics.getMatomoRequestResult($parameters, {}, 'RowEvolution'))
+#jsonResponse($analyticsResult)
```

Comment:
Would be good to wrap this code with an IF checking for ``get`` action, otherwise it's a pain to interact with this page (edit, view history, etc.).